### ISSUE TITLE
incompatibility with recent MATSim

### ISSUE THREAD

**Issue by tschlenther** [eqasim-org/matsim-sharing#9]
Since [MATSim PR 1626](https://github.com/matsim-org/matsim-libs/pull/1626), a routing module must implement `calcRoute(RoutingRequest request)` instead of `calcRoute(Facility fromFacility, Facility toFacility, double departureTime, Person person)`. This means, the current version of this contribution is incompatible with MATSim versions which are more recent than PR1626, as the `SharingRoutingModule` needs corresponding adjustments

**Comment by tschlenther** [comment]
I just realised, you probably do not want to update from stable MATSim 13.0 to some current but intermediate state, like we do in some other external contribution and project repositories...
However, i will probably have to combine matsim-sharing with recent MATSim myself.. i might have to come up with a fork and build locally.
There are more central, relevant changes in MATSim like [PR1629](https://github.com/matsim-org/matsim-libs/pull/1629/files), that will have to get taken care of as well...

**Comment by balacmi** [comment]
Yeah they are a bit of changes lately... Maybe what we can do is the following:

- we make a develop branch where you (if you have the time) can make these adaptations.
- we keep master branch stable for now, which we can also release with the MATSim 13.0 version

Would this work?

**Comment by tschlenther** [comment]
Yeah, thanks for the suggestion. If both branches would get deployed and would be usable via maven that would certainly help on the long run.
For the short term however, i consider also using maven submodules for our hamburg project, as (currently) that projects only needs recent MATsim for 1 postprocess step, only.

If i found time to contribute here, would you like me to that via PR from a fork?

Thanks again

**Comment by sebhoerl** [comment]
I would propose to move the code to matsim-libs as a contrib. @tschlenther in case you have time, feel free to move it ;) Or in any case, maybe you could coordinate with Kai if he agrees? Then I could also prepare a PR if you want

**Comment by tschlenther** [comment]
okay, i'll speak to him about it

**Comment by tschlenther** [comment]
Hi @sebhoerl i finally had the chance to talk to Kai about it: he would be happy if we could move it into matsim/contribs !
I would be super grateful if you could prepare a PR!
This would be a huge help, also for moving our hamburg project forward.
Thanks and all the best

**Comment by sebhoerl** [comment]
Ok great, I'm on holidays next week and I'll look into it after.
